packages/runtime-dom/src/directives/vShow.ts

```diff
@@ -0,0 +1,24 @@
+import { Directive } from '@vue/runtime-core'
+
+//TODO: interaction with transition module
+export const vShowGuard: Directive = {
+  beforeMount: (el, { value }) => {
+    setOriginalDisplay(el)
+    setElementDisplayByValue(el, value)
+  },
+  updated: (el, { value, oldValue }) => {
+    if (!value === !oldValue) return
+    setElementDisplayByValue(el, value)
+  },
+  beforeUnmount: el => {
+    setElementDisplayByValue(el, true)
+  }
+}
+
+function setOriginalDisplay(el: any): void {
+  el.__vOriginalDisplay = el.style.display === 'none' ? '' : el.style.display
+}
+
+function setElementDisplayByValue(el: any, value: any): void {
+  el.style.display = value ? el.__vOriginalDisplay : 'none'
+}
```

Comment:
Maybe we could use `ElementCSSInlineStyle` instead of `any`:
```ts
// or { __vOriginalDisplay: ElementCSSInlineStyle["style"]["display"] }
type ShowGuardElement = ElementCSSInlineStyle & { __vOriginalDisplay: string | null };
export const vShowGuard: Directive<ShowGuardElement>;
function setOriginalDisplay(el: ShowGuardElement): void;
function setElementDisplayByValue(el: ShowGuardElement, value: any): void;
```
What do you think about it?

Reply:
I think it is nice.
@yyx990803 ,What do you think about it?

Reply:
Yeah sounds good to me

Reply:
I already add type `ShowGuardElement `.thanks.